pkg/cmd/server/origin/master.go

```diff
@@ -471,10 +462,10 @@ func (c *MasterConfig) GetRestStorage() map[string]rest.Storage {
 	imageSignatureStorage := imagesignature.NewREST(c.PrivilegedLoopbackOpenShiftClient.Images())
 	imageStreamLimitVerifier := imageadmission.NewLimitVerifier(c.KubeClient())
 	imageStreamSecretsStorage := imagesecret.NewREST(c.ImageStreamSecretClient())
-	imageStreamStorage, imageStreamStatusStorage, internalImageStreamStorage, err := imagestreametcd.NewREST(c.RESTOptionsGetter, imageapi.DefaultRegistryFunc(defaultRegistryFunc), subjectAccessReviewRegistry, imageStreamLimitVerifier)
+	imageStreamStorage, imageStreamStatusStorage, internalImageStreamStorage, err := imagestreametcd.NewREST(c.RESTOptionsGetter, c.RegistryNameFn, subjectAccessReviewRegistry, imageStreamLimitVerifier)
```

Comment:
Which cache?  RegistryNameFn can already determine that.

On Mon, Aug 1, 2016 at 11:47 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/origin/master.go
> <https://github.com/openshift/origin/pull/8995#discussion_r73001670>:
>
> > @@ -471,10 +462,10 @@ func (c *MasterConfig) GetRestStorage() map[string]rest.Storage {
> >  	imageSignatureStorage := imagesignature.NewREST(c.PrivilegedLoopbackOpenShiftClient.Images())
> >  	imageStreamLimitVerifier := imageadmission.NewLimitVerifier(c.KubeClient())
> >  	imageStreamSecretsStorage := imagesecret.NewREST(c.ImageStreamSecretClient())
> > -	imageStreamStorage, imageStreamStatusStorage, internalImageStreamStorage, err := imagestreametcd.NewREST(c.RESTOptionsGetter, imageapi.DefaultRegistryFunc(defaultRegistryFunc), subjectAccessReviewRegistry, imageStreamLimitVerifier)
> > +	imageStreamStorage, imageStreamStatusStorage, internalImageStreamStorage, err := imagestreametcd.NewREST(c.RESTOptionsGetter, c.RegistryNameFn, subjectAccessReviewRegistry, imageStreamLimitVerifier)
>
> This should be wired up to the controller we already have that watches for
> aliases and it can block until its been primed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8995/files/04c95de82a2919295f3a4f810c6c69389c5ccf47#r73001670>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pzNO8oJxoK1YLj4yS6yo0xg0_KeGks5qbhUegaJpZM4IlDlb>
> .
>